core/src/avm2/globals/flash/system/Capabilities.as

```diff
@@ -1,20 +1,158 @@
 package flash.system {
     import __ruffle__.stub_getter;
+    import __ruffle__.stub_method;
+
     public final class Capabilities {
-        public native static function get os(): String;
-        public native static function get playerType(): String;
-        public native static function get version(): String;
-        public native static function get screenResolutionX():Number;
-        public native static function get screenResolutionY():Number;
-        public native static function get pixelAspectRatio():Number;
-        public native static function get screenDPI():Number;
+        public static function get avHardwareDisable():Boolean {
+            stub_getter("flash.system.Capabilities", "avHardwareDisable");
+            return false;
+        }
+
+        [API("665")]
+        public static function get cpuArchitecture():String {
+            stub_getter("flash.system.Capabilities", "cpuArchitecture");
+            return "x86";
+        }
+
+        public static function get hasAccessibility():Boolean {
+            stub_getter("flash.system.Capabilities", "hasAccessibility");
+            return false;
+        }
+
+        public static function get hasAudio():Boolean {
+            stub_getter("flash.system.Capabilities", "hasAudio");
+            return true;
+        }
+
+        public static function get hasAudioEncoder():Boolean {
+            stub_getter("flash.system.Capabilities", "hasAudioEncoder");
+            return true;
+        }
+
+        public static function get hasEmbeddedVideo():Boolean {
+            stub_getter("flash.system.Capabilities", "hasEmbeddedVideo");
+            return true;
+        }
+
+        public static function get hasIME():Boolean {
+            stub_getter("flash.system.Capabilities", "hasIME");
+            return true;
+        }
+
+        public static function get hasMP3():Boolean {
+            stub_getter("flash.system.Capabilities", "hasMP3");
+            return true;
+        }
+
+        public static function get hasPrinting():Boolean {
+            stub_getter("flash.system.Capabilities", "hasPrinting");
+            return true;
```

Comment:
Probably makes sense to be true, but since we don't yet support the PrintJob class (not even stubbing the isSupported flag to false) I imagine content that checks this will just fail later as I assume this would be checked for the purpose of safe-guarding PrintJob code.

Reply:
That makes sense, fixed!